tests/e2e/statefulsets.go

```diff
@@ -422,6 +424,151 @@ var _ = ginkgo.Describe("[csi-block-vanilla] [csi-supervisor] statefulset", func
+		gomega.Expect(len(ssPodsBeforeScaleDown.Items) == int(replicas)).To(gomega.BeTrue(), "Number of Pods in the statefulset should match with number of replicas")
+
+		ginkgo.By("Verify all volumes are attached to Nodes after Statefulsets is scaled up")
+		// Get the list of Volumes attached to Pods before scale down
+		//var volumesBeforeScaleDown []string
+		for _, sspod := range ssPodsBeforeScaleDown.Items {
+			_, err := client.CoreV1().Pods(namespace).Get(ctx, sspod.Name, metav1.GetOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			for _, volumespec := range sspod.Spec.Volumes {
+				if volumespec.PersistentVolumeClaim != nil {
+					pvclaimName := volumespec.PersistentVolumeClaim.ClaimName
+
+					pvclaim, err := client.CoreV1().PersistentVolumeClaims(namespace).Get(ctx, pvclaimName, metav1.GetOptions{})
+					gomega.Expect(pvclaim).NotTo(gomega.BeNil())
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					pv := getPvFromClaim(client, statefulset.Namespace, volumespec.PersistentVolumeClaim.ClaimName)
+
+					ginkgo.By("Expanding current pvc")
+					sizeBeforeexpansion := pvclaim.Status.Capacity[v1.ResourceStorage]
+					pvcSizeBeforeExpansion, _ = sizeBeforeexpansion.AsInt64()
+					framework.Logf("pvcsize : %d", pvcSizeBeforeExpansion)
+					currentPvcSize := pvclaim.Spec.Resources.Requests[v1.ResourceStorage]
+					newSize := currentPvcSize.DeepCopy()
+					newSize.Add(resource.MustParse("1Gi"))
+					framework.Logf("currentPvcSize %v, newSize %v", currentPvcSize, newSize)
+					pvclaim, err = expandPVCSize(pvclaim, newSize, client)
+					framework.ExpectNoError(err, "While updating pvc for more size")
+					gomega.Expect(pvclaim).NotTo(gomega.BeNil())
+
+					ginkgo.By("Waiting for file system resize to finish")
+					_, err = waitForFSResize(pvclaim, client)
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+					err = verifyVolumeMetadataInCNS(&e2eVSphere, pv.Spec.CSI.VolumeHandle, volumespec.PersistentVolumeClaim.ClaimName, pv.ObjectMeta.Name, sspod.Name)
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				}
+			}
+		}
+		ginkgo.By("Delete statefulset with cascade = false")
+		cascade := false
+		err = client.AppsV1().StatefulSets(namespace).Delete(context.TODO(), statefulset.Name, metav1.DeleteOptions{OrphanDependents: &cascade})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		statefulset = GetResizedStatefulSetFromManifest(namespace)
+		ginkgo.By("Creating statefulset")
+		CreateStatefulSet(namespace, statefulset, client)
+		replicas = *(statefulset.Spec.Replicas)
+
+		incresedReplicaCount := replicas + 1
+		ginkgo.By(fmt.Sprintf("Scaling up statefulsets to number of Replica: %v", incresedReplicaCount))
+		_, scaleupErr := fss.Scale(client, statefulset, incresedReplicaCount)
+		gomega.Expect(scaleupErr).NotTo(gomega.HaveOccurred())
+		fss.WaitForStatusReplicas(client, statefulset, incresedReplicaCount)
+		fss.WaitForStatusReadyReplicas(client, statefulset, incresedReplicaCount)
+
```

Comment:
Scaleup of replica count is a necessary step?? aren't the default replicas reflect the increased capacity directly?

Reply:
You mean to say the existing PVCs and default replica count pods do not reflect the increased capacity of PVCs until the scaleup is not done ?? This gives an impression that only scaleup operation will make the volume expansion work.

If not, you need to confirm the increased capacity of the PVCs and with the default replica count first and as an additional step - scale up the replica and confirm the new PVC's are also having increased capacity on PVC.